src/bio/bio_device.c

```diff
@@ -688,115 +809,69 @@ bio_set_led_state(struct bio_xs_context *xs_ctxt, uuid_t dev_uuid,
 		return -DER_NONEXIST;
 	}
 
+	/* Init LED context state */
+	opts.all_devices = false;
+	opts.finished = false;
+	opts.led_state = SPDK_VMD_LED_STATE_UNKNOWN;
+	opts.status = 0;
+
 	/* LED will be reset to the original saved state */
+	opts.action = LED_ACTION_SET;
 	if (reset) {
-		new_led_state = bio_dev->bb_led_state;
-		D_GOTO(skip_led_str, rc = 0);
+		opts.action = LED_ACTION_RESET;
+		opts.led_state = bio_dev->bb_led_state;
+		goto skip_led_str;
 	}
 
-	if (led_state == NULL)
+	if (led_state_str == NULL)
 		return -DER_INVAL;
 
 	/* Determine SPDK LED state based on led_state string */
-	if (strcasecmp(led_state, "identify") == 0) {
-		new_led_state = SPDK_VMD_LED_STATE_IDENTIFY;
-	} else if (strcasecmp(led_state, "on") == 0) {
-		new_led_state = SPDK_VMD_LED_STATE_FAULT;
-	} else if (strcasecmp(led_state, "fault") == 0) {
-		new_led_state = SPDK_VMD_LED_STATE_FAULT;
-	} else if (strcasecmp(led_state, "off") == 0) {
-		new_led_state = SPDK_VMD_LED_STATE_OFF;
-	} else {
+	for (led_state = SPDK_VMD_LED_STATE_OFF;
+	     led_state <= SPDK_VMD_LED_STATE_REBUILD;
+	     led_state++) {
+		if (strcmp(led_state_str, g_led_states[led_state]) == 0) {
+			opts.led_state = (enum spdk_vmd_led_state)led_state;
+			break;
+		}
+	}
+
+skip_led_str:
+	if (opts.led_state == SPDK_VMD_LED_STATE_UNKNOWN) {
 		D_ERROR("LED state is not valid or supported\n");
 		return -DER_NOSYS;
```

Comment:
seems like this one should be -DER_INVAL

Reply:
done